src/link/resolvePeers.ts

```diff
@@ -99,6 +101,11 @@ export default function (
   })
 
   R.values(resolvedTree).forEach(node => {
+    const nodeMajor = process.version.substring(0, process.version.indexOf('.'))
+    if (node.sideEffectsCache && node.sideEffectsCache[nodeMajor]) {
+      node.isBuilt = true
+      node.centralLocation = node.sideEffectsCache[nodeMajor]
```

Comment:
assign this two values right where the node is created: https://github.com/pnpm/supi/blob/b7dae8b25c3ae1e2575e1b2c3460c971a4d88533/src/link/resolvePeers.ts#L182

it is much easier to read the code when all properties of the object are set in one place and stay immutable